src/ResourceManager/EventGrid/Commands.EventGrid/Topic/UpdateAzureEventGridTopic.cs

```diff
@@ -0,0 +1,133 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.EventGrid.Models;
+using Microsoft.Azure.Commands.EventGrid.Utilities;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.EventGrid.Models;
+
+namespace Microsoft.Azure.Commands.EventGrid
+{
+    [Cmdlet(VerbsData.Update, EventGridTopicVerb, SupportsShouldProcess = true, DefaultParameterSetName = TopicNameParameterSet), OutputType(typeof(PSTopic))]
+    public class UpdateAzureEventGridTopic : AzureEventGridCmdletBase
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "Resource Group Name.",
+            ParameterSetName = TopicNameParameterSet)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasResourceGroup)]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = "EventGrid Topic Name.",
+            ParameterSetName = TopicNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [Alias("TopicName")]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "EventGrid Topic ResourceID.",
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "EventGrid Topic object.",
+            ParameterSetName = TopicInputObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSTopic InputObject { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            Position = 2,
+            HelpMessage = "Hashtable which represents resource Tags.",
+            ParameterSetName = TopicNameParameterSet)]
+        [Parameter(
+            Mandatory = true,
+            Position = 1,
+            HelpMessage = "Hashtable which represents resource Tags.",
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [Parameter(
+            Mandatory = true,
+            Position = 1,
+            HelpMessage = "Hashtable which represents resource Tags.",
+            ParameterSetName = TopicInputObjectParameterSet)]
+        public Hashtable Tag { get; set; }
```

Comment:
@kalyanaj in the future, will there be any other properties that the user can update for a Topic? I'm wondering if we should make `Tag` optional so when a new property is added, we don't have to (1) make it optional to avoid breaking changes, and (2) have `Tag` be the only mandatory property parameter.

Reply:
@cormacpayne That's a good point. Currently, there's no other property but we don't know how it can evolve, so I have made Tag an optional parameter per your suggestion above and have regenerated the help content as well.

Reply:
@cormacpayne , if the changes look good, can you please help merge this PR?  Would like to get this checked in this week.